tsdb/exemplar.go

```diff
@@ -169,6 +175,18 @@ func (ce *CircularExemplarStorage) AddExemplar(l labels.Labels, e exemplar.Exemp
 	ce.lock.Lock()
 	defer ce.lock.Unlock()
 
+	// Exemplar label length does not include chars involved in text rendering such as quotes
```

Comment:
This will silently miss the exemplars as this is called on Commit(). Since this is a validation, this should go in the `AppendExemplar` of the appender and reject this exemplar early.

Reply:
Currently, without some of the changes we will potentially make in #8296, all exemplar validation is done here in `AddExemplar`. If anything, this check should be done by the OpenMetrics parser, though we'd probably still want this and other validation checks in other spots as well now that data can be remote written to Prometheus itself.

Reply:
> Currently, without some of the changes we will potentially make in #8296

Let's wait for it then :) this PR should be a small change after it

Reply:
And metric would be necessary in head.go to detect such silent dropping of exemplars